snippets/csharp/System.Net.Sockets/Socket/Overview/socket.cs

```diff
@@ -3,97 +3,107 @@
-
-            tempSocket.Connect(ipe);
-
-            if(tempSocket.Connected)
-            {
-                s = tempSocket;
-                break;
-            }
-            else
-            {
-                continue;
-            }
-        }
-        return s;
+        Uri? uri = args.Any() ? new Uri(args[0]) : null;
+
+        // Sync:
+        SendHttpRequest(uri);
+
+        // Async:
+        await SendHttpRequestAsync(uri);
     }
 
-    // This method requests the home page content for the specified server.
-    private static string SocketSendReceive(string server, int port)
+//<Snippet1>
+private static void SendHttpRequest(Uri? uri = null, int port = 80)
+{
+    uri ??= new Uri("http://example.com");
+
+    // Construct a minimalistic HTTP/1.1 request
+    byte[] requestBytes = Encoding.ASCII.GetBytes(@$"GET {uri.AbsoluteUri} HTTP/1.1
+Host: {uri.Host}
+Connection: Close
+
+");
+
+    // Create and connect a dual-stack socket
+    using Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+    socket.Connect(uri.Host, port);
+
+    // Send the request
+    int sendCnt = socket.Send(requestBytes);
+
+    // Do minimalistic buffering assuming ASCII response
+    byte[] responseBytes = new byte[256];
+    char[] responseChars = new char[256];
+
+    while (true)
     {
-        string request = "GET / HTTP/1.1\r\nHost: " + server +
-            "\r\nConnection: Close\r\n\r\n";
-        Byte[] bytesSent = Encoding.ASCII.GetBytes(request);
-        Byte[] bytesReceived = new Byte[256];
-        string page = "";
-
-        // Create a socket connection with the specified server and port.
-        using(Socket s = ConnectSocket(server, port)) {
-
-            if (s == null)
-                return ("Connection failed");
-
-            // Send request to the server.
-            s.Send(bytesSent, bytesSent.Length, 0);
-
-            // Receive the server home page content.
-            int bytes = 0;
-            page = "Default HTML page on " + server + ":\r\n";
-
-            // The following will block until the page is transmitted.
-            do {
-                bytes = s.Receive(bytesReceived, bytesReceived.Length, 0);
-                page = page + Encoding.ASCII.GetString(bytesReceived, 0, bytes);
-            }
-            while (bytes > 0);
-        }
-
-        return page;
+        int byteCount = socket.Receive(responseBytes);
+
+        // Receiving 0 bytes means EOF has been reached
+        if (byteCount == 0) break;
+
+        // Convert byteCount bytes to ASCII characters using the 'responseChars' buffer as destination
+        int charCount = Encoding.ASCII.GetChars(responseBytes, 0, byteCount, responseChars, 0);
+
+        // Print the contents of the 'responseChars' buffer to Console.Out
+        Console.Out.Write(responseChars, 0, charCount);
     }
 
-    public static void Main(string[] args)
+    Console.Out.Flush();
```

Comment:
IIRC console writer automatically flushes, so this is not needed. Same for async example